### PR TITLE
merge Bitcoin #16325 #16210 #15784 #15757 : Backport

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
64 bit Test Failed due to Disk space 
    print("{0} {1: <5} {2} {3}".format(colors[event.source.rstrip()], event.source, lines[0], colors["reset"]))
**OSError: [Errno 28] No space left on device**
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
OSError: [Errno 28] No space left on device,  please retrigger once the space is reclaimed

**Comment by vijaydasmp** [comment]
Hello @UdjinM6  requesting review

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta , Requesting review

File: doc/psbt.md
Comment (UdjinM6) [inline doc/psbt.md:85]: 15757: we have no `analyzepsbt`

File: src/rpc/blockchain.cpp
Comment (UdjinM6) [inline src/rpc/blockchain.cpp:50]: 16325: changes in `include`s do not match. also, missing changes in `test/functional/wallet_resendwallettransactions.py`.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/rpc/blockchain.cpp
Comment (vijaydasmp) [reply src/rpc/blockchain.cpp]: For header file changes 
we cant remove this **#include <key_io.h>** yet, as  this will cause build to break

In file included from ./script/descriptor.h:9,
                 from rpc/blockchain.cpp:24:
./script/sign.h:14:7: note: forward declaration of ‘class CKey’
   14 | class CKey;

To remove it we need to implement Merge #16227: Refactor CWallet's inheritance chain 

Also moving of #include <condition_variable> header from line 47 to 44 in original change looks different in our dash code as we don't have line #include <memory> in line 45 (cant pull this changes as this is marked as DNM)  https://github.com/bitcoin/bitcoin/pull/16325/files src/rpc/blockchain.cpp
So header changes are looking different

Somehow I missed this test/functional/wallet_resendwallettransactions.py I have added it now
I have marked 16325 as partial Merge we can remove key_io once Merge #16227 is merged

File: doc/psbt.md
Comment (vijaydasmp) [reply doc/psbt.md]: Fixed it

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

nit: please name 16325 as not being partial (all substance was done), but that's not a blocker imo

**Comment by vijaydasmp** [comment]
Sure, will do

**Comment by vijaydasmp** [comment]
Hello @UdjinM6  please re approve , approval got dismissed after force push after rebase

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**
